packages/react-reconciler/src/ReactFiberCommitWork.new.js

```diff
@@ -1589,6 +1609,30 @@ function getHostSibling(fiber: Fiber): ?Instance {
   }
 }
 
+function commitSingletonPlacement(finishedWork: Fiber): void {
+  if (!supportsMutation) {
+    return;
+  }
+  if (enableFloat) {
+    // We get back the singleton Instance from placement and need to capture it as the stateNode
+    // for the Fiber.
+    commitSingletonPlacementImpl(
+      finishedWork.type,
+      finishedWork.memoizedProps,
+      finishedWork.stateNode,
+      finishedWork,
+    );
+    const parent = finishedWork.stateNode;
+    const before = getInsertionEdge(parent);
+    let child = finishedWork.child;
+    while (child !== null) {
+      insertOrAppendPlacementNode(child, before, parent);
```

Comment:
Instead of doing this traversal here, you can do the same thing that HostPortal does.

https://github.com/facebook/react/blob/aa80a309b19d59832ffeb82dfb6235da0c93dd42/packages/react-reconciler/src/ReactFiberBeginWork.new.js#L3316-L3326

It uses a hack to ensure that the initial mount reconciliation pass tags the children of the Portal with Placement. This is similar since it's effectively a Portal into the existing node.

Then you get the same pass as when new children are added.